concepts/methods/about.md

```diff
@@ -0,0 +1,134 @@
+Everything within this `impl` block will be associated with the `Wizard` type.
+Note that a struct is allowed to have multiple `impl` blocks.
+
+### Defining method with `&self`
+
+If we wish to implement an `info` method to display the basic information of the wizard,
+we could define this method inside an `impl` block for `Wizard`:
+
+```rust
+impl Wizard {    
+    fn info(&self) {
+        println!("{} studies in House {} at Hogwarts", self.name, self.house);
+    }
+}
+```
+
+Note that in the signature for `info`, we use `&self`, which is short for `self: &Self`.
+The type `Self` is an alias for the type that the `impl` block is for, in this case `Wizard`.
+
+Recall that we mentioned that a method must have a parameter named `self` of type `Self` as its first parameter,
+so Rust allows us to abbreviate this with only the name `self` in the first parameter spot for convenience.
+
+Note that we still need to use the `&` in front of the `self` shorthand to indicate this method
+borrows the `Self` instance so that we could read the `name` and `house` fields of a `Wizard` instance.
+
+With the `info` method implemented, we could read a wizard's basic information:
+```rust
+let harry = Wizard {
+  name: String::from("Harry Potter"),
+  house: String::from("Gryffindor"),
+};
+
+harry.info();
+// => Harry Potter studies in House Gryffindor at Hogwarts
+```
+
+### Defining method with `&mut self`
+
+Methods can borrow `self` immutably, as in `&self` for the signature of the method `info`).
+However, if we wish to implement a method that mutates the fields of a struct instance, we must borrow the instance mutably.
+
+Let's say we want to set up a new rule at Hogwarts to allow students to switch houses.
+We could implement a method `switch_house` as following:
+
+```rust
+impl Wizard {
+  fn switch_house(&mut self, new_house: String) {
+    self.house = new_house;
+  }
+}
+```
+
+To enable a method to borrow `self` mutably, we simply use `&mut self` in the signature for the method.
+Now we can switch houses for a wizard who studies at Hogwarts:
+
+```rust
+// Note that the variable `harry` must be mutable
+let mut harry = Wizard {
+  name: String::from("Harry Potter"),
+  house: String::from("Gryffindor"),
+};
+
+harry.info();
+// => Harry Potter studies in House Gryffindor at Hogwarts
+harry.switch_house(String::from("Slytherin"));
+harry.info()
+// => Harry Potter studies in House Slytherin at Hogwarts
+```
+
+### Multiple 'impl' blocks
+
+In Rust, each struct is allowed to have multiple `impl` blocks.
+So we can group all methods in the same `impl` block or have multiple `impl` blocks for different groups of methods.
+Both approaches are valid syntax.
+
+## Associated Functions
+
+All functions that are defined within an `impl` block are called associated functions.
+Some associated functions for a struct don't necessarily need to have `self` as their first parameter
+(and thus are not methods).
```

Comment:
For me it's actually from [the current rust-lang book](https://doc.rust-lang.org/book/ch05-03-method-syntax.html#associated-functions):

> All functions defined within an impl block are called associated functions because they’re associated with the type named after the impl. We can define associated functions that don’t have self as their first parameter (and thus are not methods) because they don’t need an instance of the type to work with.

I don't have a strong preference here as I like your edits as well.

Reply:
Interesting, TIL. You're right that methods are officially a subset of associated functions. If you don't feel strongly, though, I do still prefer my phrasing.